### PR TITLE
Network trees

### PR DESCRIPTION
**Pull request by szok**
TODO
- [x] breadcrumps
- [x] sql performance

zrzut ekranu 2016-03-23 o 09 56 11

### CODE REVIEW

File: src/ralph/networks/admin.py [hunk 1]
```diff
@@ -45,11 +48,42 @@ class Meta:
 
 
 @register(Network)
-class NetworkAdmin(RalphMPTTAdmin):
+class NetworkAdmin(RalphAdmin):
+
+    def subnetwork(self, obj):
```
Comment (mkurek) [inline src/ralph/networks/admin.py]: please move these fields under attributes definition

File: src/ralph/networks/admin.py [hunk 2]
```diff
@@ -45,11 +48,42 @@ class Meta:
...
+                urlencode({'parent': obj.pk}),
+                _('Show'),
+                count
+            )
+        return ''
+
+    subnetwork.short_description = 'Subnetwork'
+    subnetwork.allow_tags = True
+
+    def name(self, obj):
```
Comment (mkurek) [inline src/ralph/networks/admin.py]: do we really need this field declaration? why default-way is not working (just declaring `'name'` on `list_display`)?

Comment (szok) [inline src/ralph/networks/admin.py]: to delete this.

File: src/ralph/networks/templates/networks/network_change_list.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{% extends 'admin/change_list.html' %}
+{% block breadcrumbs %}
+<ul class="breadcrumbs">
+  <li><a href="{% url 'admin:networks_network_changelist' %}">Network</a></li>
```
Comment (mkurek) [inline src/ralph/networks/templates/networks/network_change_list.html:4]: this is not the original behaviour of https://github.com/allegro/ralph/blob/ng/src/ralph/admin/templates/admin/breadcrumbs_foundation.html - couldn't you wrap the for loop here in some block and overwrite it with calling super and adding parent networks after default content?

File: src/ralph/networks/admin.py [hunk 4]
```diff
@@ -90,6 +124,18 @@ def changeform_view(
...
         )
 
+    def changelist_view(self, request, extra_context=None):
+        if extra_context is None:
+            extra_context = {}
+        parent = request.GET.get('parent', None)
+        if parent:
+            parent = Network.objects.get(pk=parent)
+            extra_context.update(
+                {'parents_network': parent.get_ancestors(include_self=True)}
```
Comment (mkurek) [inline src/ralph/networks/admin.py:109]: are there ancestors returned in specified order?

**Comment by mkurek** [comment]
:ok:

**Comment by ghost** [comment]
:ok:

**Pull request merged**